### DISCUSSION TITLE
How to process messages, commands, buttons - in one class?

### DISCUSSION THREAD

**Discussion by ss-pt** [westacks/telebot#64] [Q&A] [answered]
Let's say I want to receive any messages, button clicks, etc. in one place. How can I do it?

**config/telebot.php**

```php
'handlers' => [
    App\Services\TeleBotHandler::class,
],
```

**app/Services/TeleBotHandler.php**

```php
<?php

use App\Services;

use WeStacks\TeleBot\Interfaces\UpdateHandler;
use WeStacks\TeleBot\Objects\Update;
use WeStacks\TeleBot\TeleBot;

class TeleBotHandler extends UpdateHandler
{
    public static function trigger(Update $update, TeleBot $bot): bool
    {
        return true;
    }

    public function handle()
    {
        $update = $this->update;
        $bot = $this->bot;

        // chat_id => $this->update->message->chat->id
        return $this->sendMessage([
            'text' => 'Hello, World!',
            'chat_id' => $this->update->message->chat->id,
        ]);
    }
}
```

it doesn't work, I get the error:

`[2023-01-16 14:57:04] production.ERROR: Class "WeStacks\TeleBot\Interfaces\UpdateHandler" not found {"exception":"[object] (Error(code: 0): Class \"WeStacks\\TeleBot\\Interfaces\\UpdateHandler\" not found at /Users/htdocs/test/app/Services/TeleBotHandler.php:9)
`

**Comment by punyflash** [comment]
Firstly, I'd not recommend you doing everything in one class, because you will grow yourself a "super class" and it will be impossible to maintain it. Please check out [SOLID](https://en.wikipedia.org/wiki/SOLID) principles.

Secondly, check out #62 to resolve your error.

Thirdly, in your example, you already have a class that will process any update, because `trigger` method returns true no matter what.

**Comment by ss-pt** [comment, marked as answer]
Thanks, I got it. I made several mistakes!

```php
<?php

namespace App\Services\BotHandlers;

use WeStacks\TeleBot\Handlers\UpdateHandler;

class TeleBotHandler extends UpdateHandler
{
    public function trigger(): bool
    {
        return true;
    }

    public function handle()
    {
        $update = $this->update;
        $bot = $this->bot;
        
        return $this->sendMessage([
            'text' => var_export($this->update->message->chat, true),
            'chat_id' => $this->update->message->chat->id,
        ]);
    }
}
```

**Reply by ss-pt** [reply]
@punyflash **And how can I catch a button click in this class?**

```php
<?php

namespace App\Services\BotHandlers;

use WeStacks\TeleBot\Handlers\UpdateHandler;

class TeleBotHandler extends UpdateHandler
{
    public function trigger(): bool
    {
        return true;
    }

    public function handle()
    {
        $update = $this->update;

        return $this->sendMessage([
            'text' => 'Click button:',
            'chat_id' => $this->update->message->chat->id,
            'reply_markup'   =>  [
                'inline_keyboard' => [[
                    ['text' => 'answer1', 'callback_data' => 'answer1cb' ],
                    ['text' => 'answer2', 'callback_data' => 'answer2cb' ],
                ]]
            ],
        ]);
    }
}
```

Now when the button is pressed, nothing happens!

**Reply by punyflash** [reply]
Read the docs: https://core.telegram.org/bots/api#callbackquery, I'll not write bot for you. Just write your logic in `handle` method. Every update is going through it.
